ios/engine/KMEI/KeymanEngine/Classes/Settings/SettingsViewController.swift

```diff
@@ -161,13 +200,28 @@ open class SettingsViewController: UITableViewController {
         }
       case "installfile":
         cell.accessoryType = .disclosureIndicator
+      case "forcederror":
+        cell.accessoryType = .disclosureIndicator
       default:
         log.error("unknown cellIdentifier(\"\(cellIdentifier ?? "EMPTY")\")")
         cell.accessoryType = .none
     }
     
     return cell
   }
+
+  @objc func reportingSwitchValueChanged(_ sender: Any) {
+    let userData = Storage.active.userDefaults
+    if let toggle = sender as? UISwitch {
+      userData.set(toggle.isOn, forKey: Key.optShouldReportErrors)
+      SentryManager.enabled = toggle.isOn
+
+      userData.synchronize()
+    }
+
+    // Forward the thing to KMW; actually USE that value.
+    Manager.shared.inputViewController.refreshCrashReporting()
```

Comment:
> I am not 100% confident that this is correct but I wanted to suggest it because it raises a few questions in my mind:
> 
>     1. It makes me uncomfortable having the settings controller setting the state for both crash reporting mechanisms in different ways, so ...
> 
>     2. Why isn't `SentryManager` responsible for reading its crash reporting state just like `inputViewController` is? (i.e. why can't we call `SentryManager.refreshCrashReporting()`.

It goes back to this section of the code:  https://github.com/keymanapp/keyman/pull/4106#discussion_r541986386

The main way we're able to enable and disable error reporting requires use of these two code sections:

https://github.com/keymanapp/keyman/blob/f5fc278aa672d5f390f79df0d96f373477da71d8/ios/engine/KMEI/KeymanEngine/Classes/Errors/SentryManager.swift#L53-L58

https://github.com/keymanapp/keyman/blob/f5fc278aa672d5f390f79df0d96f373477da71d8/ios/engine/KMEI/KeymanEngine/Classes/Errors/SentryManager.swift#L70-L79

We pretty much have to do it on something central to the `SentryManager` itself, hence use of the class's use of `static` typing on related functionality.

The problem is that `SentryManager` is part of the `KeymanEngine` so that the engine is capable of logging error information, while the actual UI and actual Sentry setup is done within Keyman app code.

>     3. If we want to avoid `SentryManager` having to access `userData`, then it seems like we should have a wrapper function `Manager.shared.refreshCrashReporting()` that looks after refreshing both  `inputViewController` and `SentryManager`?

If other people wish to use the framework, how would _they_ wish to enable and disable error reporting if they made use of Sentry as well?  It'd be pretty awkward of us to forcibly prescribe use of an arbitrary key within `userData` on other framework consumers.  Moving this check to the suggested `Manager.shared.refreshCrashReporting` does little to alleviate this issue unless we give _this version_ a parameter.  Which still keeps the method signatures distinct.

> 
>     4. Or something like that. I don't have a totally clear picture of the state machines and structures involved here. Understand it may be difficult to do that in the same way during initialisation.
> 
>     5. I reordered the code to make it clearer in my mind. I think that order is okay, isn't it? We shouldn't need to do `Manager.shared.inputViewController.refreshCrashReporting()` if we can't actually get an updated value for the toggle?

That function is used to ensure that the error-reporting state is correctly passed through to the KeymanWeb engine, which requires a JS call into the WebView.  Naturally, Web code can't access Swift-side properties directly, and constantly polling for its state asynchronously is a bad idea, given that it's part of the control logic for error reporting.

If anything... I think this code block should be changed:

https://github.com/keymanapp/keyman/blob/f5fc278aa672d5f390f79df0d96f373477da71d8/ios/engine/KMEI/KeymanEngine/Classes/KeymanWebViewController.swift#L363-L368

It should refer to `SentryManager.enabled`, rather than the `userData` property.  That'd probably be the cleaner approach, with the `refresh__` method used to actively tell the keyboard to re-check the value.

Reply:
> If other people wish to use the framework, how would _they_ wish to enable and disable error reporting if they made use of Sentry as well? It'd be pretty awkward of us to forcibly prescribe use of an arbitrary key within `userData` on other framework consumers.

This is true. I see you do already have a reference to that `userData` key in Keyman Engine as you note a few lines later, in: https://github.com/keymanapp/keyman/blob/f5fc278aa672d5f390f79df0d96f373477da71d8/ios/engine/KMEI/KeymanEngine/Classes/KeymanWebViewController.swift#L365

So fixing that requires a more substantial refactor, as you need to pull those existing references to the userData out of Keyman Engine and into Keyman itself as well.

> constantly polling for its state asynchronously is a bad idea

Well yes.

> If anything... I think this code block should be changed:

That does about half of it. It doesn't fix the need for the app-side code to make two calls to refresh both Swift-side and web-side crash reporting enabled state.

Reply:
The two most-recent commits are largely focused on this comment chain and the `silenced` -> `_enabled` requests.  The results should feel a lot more coherent, though the changes are distinct from the implementation specifics suggested here.